Python/ceval.c

```diff
@@ -3156,18 +3156,19 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
         case TARGET(BEFORE_ASYNC_WITH): {
             _Py_IDENTIFIER(__aexit__);
             _Py_IDENTIFIER(__aenter__);
-
             PyObject *mgr = TOP();
-            PyObject *exit = special_lookup(tstate, mgr, &PyId___aexit__),
-                     *enter;
+            PyObject *enter = special_lookup(tstate, mgr, &PyId___aenter__);
             PyObject *res;
-            if (exit == NULL)
+            if (enter == NULL) {
+                goto error;
+            }
+            PyObject *exit = special_lookup(tstate, mgr, &PyId___aexit__);
+            if (exit == NULL) {
+                Py_DECREF(enter);
```

Comment:
Huh, the old code had a refleak in the error case where aexit existed, but aenter was missing.

Reply:
@ncoghlan Yes!